### PR TITLE
OV-29: Add Video Items Menu

### PR DESCRIPTION
**Pull request by lfelix3011**
This Pull Request is for task #29 

- [x] Add a menu that can apply any type of body as a child
- [x] After you click on menu icon, menu body should open
- [x] Menu body should apply child and title as a props

As suggested in review uses of icons of font-awsome
image

### CODE REVIEW

File: frontend/src/bundles/video-editor/components/menu-body/menu-body.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                        borderRadius: '8px',
+                        boxShadow: 'lg',
+                        bg: 'background.900',
+                        color: 'white',
+                    }}
+                >
+                    <Flex justifyContent="space-between" alignItems="center">
+                        <Heading as="h3" size="md">
+                            {title}
+                        </Heading>
```
Comment (o-nedashkivska) [inline frontend/src/bundles/video-editor/components/menu-body/menu-body.tsx:44]: You should add variant to the heading. We have already merged business styles into next.

File: frontend/src/bundles/common/components/icons/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { createIcon } from '../components.js';
+
+const TemplatesIcon = createIcon({
+    displayName: 'Templates',
+    viewBox: '0 0 18 18',
+    d: 'M2 18c-.55 0-1.02-.196-1.413-.587A1.926 1.926 0 010 16V2C0 1.45.196.98.588.587A1.926 1.926 0 012 0h14c.55 0 1.02.196 1.413.588C17.803.979 18 1.45 18 2v14c0 .55-.196 1.02-.587 1.413A1.926 1.926 0 0116 18H2zm14-5H2v3h14v-3zm-2.5-2H16V2h-2.5v9zM2 11h2.5V2H2v9zm4.5 0h5V2h-5v9z',
+});
```
Comment (o-nedashkivska) [inline frontend/src/bundles/common/components/icons/icons.tsx:7]: For icons, it is better to use @chakra-ui/icons. If it doesn’t have the necessary icon, you can use https://github.com/FortAwesome. In [this pr](https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/36) you can see the example

File: frontend/src/bundles/common/helpers/utilities.ts
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/helpers/utilities.ts:1]: please split them by folders to follow the project structure

File: frontend/src/bundles/video-editor/components/menu/menu.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import { Box, Flex, VStack } from '~/bundles/common/components/components.js';
+import {
+    isEmptyArray,
+    isNullOrUndefined,
+} from '~/bundles/common/helpers/helpers.js';
+import { useCallback } from '~/bundles/common/hooks/hooks.js';
+
+import { type MenuItem } from '../../types/types.js';
+
+type MenuProperties = {
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/video-editor/components/menu/menu.tsx:10]:
```suggestion
type Properties = {
```

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/icons/icons.tsx (hunk 2 above)
Comment (lfelix3011) [reply frontend/src/bundles/common/components/icons/icons.tsx:7]: @o-nedashkivska  @anton-otroshchenko  Hi, thanks for all your feedbacks, really apreciate it, i dont see the same icons neither in @chakra-ui/icons or in [Fort](https://github.com/FortAwesome)

I suggested using the more similar, what do you think:
image

File: frontend/src/bundles/common/helpers/utilities.ts
Comment (lfelix3011) [reply frontend/src/bundles/common/helpers/utilities.ts:1]: @anton-otroshchenko Thanks, do you mean a folder for utilities or divide the 3 function in separate folders ???

File: frontend/src/bundles/common/components/icons/icons.tsx (hunk 2 above)
Comment (o-nedashkivska) [reply frontend/src/bundles/common/components/icons/icons.tsx:7]: Looks good to me :)

File: frontend/src/bundles/common/helpers/utilities.ts
Comment (WilsonBravo) [reply frontend/src/bundles/common/helpers/utilities.ts:1]: I think is probably following the next structure:

    helpers/
    │
    ├── helpers.ts
    │
    ├── is-null-or-undefined/
    │   └── is-null-or-undefined.ts
    │
    ├── is-string-null-or-empty/
    │   └── is-string-null-or-empty.ts
    │
    └── other-helper/
        └── other-helper.ts

File: frontend/src/bundles/video-editor/components/menu-body/menu-body.tsx (hunk 1 above)
Comment (lfelix3011) [reply frontend/src/bundles/video-editor/components/menu-body/menu-body.tsx:44]: @o-nedashkivska Thanks, i am already using the variant h3, after reviewing the figma dsesing i remove the size property as we dont need it.

File: frontend/src/bundles/common/helpers/utilities.ts
Comment (lfelix3011) [reply frontend/src/bundles/common/helpers/utilities.ts:1]: @anton-otroshchenko , @WilsonBravo @o-nedashkivska  thanks fot the feedback, so as array and string are common things that need method to abstract actions working with them i put it like the structures and image below, please tell me if this is ok or if we can improve it

helpers/
│
├── helpers.ts
│
├── array-helper/
│   └── array-helper.ts
│   └── is-empty-array.ts
│
├── is-null-or-undefined/
│   └── is-null-or-undefined.ts
│
├── string-helper/
│   └── string-helper.ts
│   └── is-null-or-undefined.ts


image

Comment (anton-otroshchenko) [reply frontend/src/bundles/common/helpers/utilities.ts:1]: I think this structure fits the best

helpers/check-is-array-empty/check-is-array-empty.ts
helpers/check-is-null-or-undefined/check-is-null-or-undefined.ts
helpers/check-is-string-null-or-empty/check-is-string-null-or-empty.ts

and you will just reexport them in helpers.ts

File: frontend/src/bundles/common/components/components.ts [hunk 4]
```diff
@@ -12,18 +12,29 @@ export {
...
     FormErrorMessage,
     Heading,
+    Icon,
     IconButton,
     InputGroup,
     InputRightElement,
     SimpleGrid,
     Text,
     VStack,
 } from '@chakra-ui/react';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/components.ts:29]: common/icons/icons.ts

File: frontend/src/bundles/common/helpers/array-helper/is-empty-array.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/helpers/array-helper/is-empty-array.ts:1]: why we need this helper?

`arr.length === 0`
or
`!arr.length`

if you have undefined value
`arr?.length === 0`

File: frontend/src/bundles/common/helpers/is-null-or-undefined/is-null-or-undefined.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/helpers/is-null-or-undefined/is-null-or-undefined.ts:1]: `null/undefine ?? fallbackValue`

File: frontend/src/bundles/common/helpers/string-helpers/is-null-or-undefined.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/helpers/string-helpers/is-null-or-undefined.ts:1]: the same

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/helpers/array-helper/is-empty-array.ts
Comment (lfelix3011) [reply frontend/src/bundles/common/helpers/array-helper/is-empty-array.ts:1]: @nikita-remeslov thanks for the feedback, then i remove those helper and check by nullish values "!items || items.length === 0 || "

File: frontend/src/bundles/common/helpers/is-null-or-undefined/is-null-or-undefined.ts
Comment (lfelix3011) [reply frontend/src/bundles/common/helpers/is-null-or-undefined/is-null-or-undefined.ts:1]: @nikita-remeslov thanks for the feedback, then i remove those helper and check by nullish values "!items"

File: frontend/src/bundles/common/enums/app-route.enum.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ const AppRoute = {
     SIGN_IN: '/sign-in',
     SIGN_UP: '/sign-up',
     STUDIO: '/studio',
+    VIDEO_EDITOR: '/video-editor',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/enums/app-route.enum.ts:6]: why we need this route?

File: frontend/src/bundles/common/icons/icons-map.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { DownloadIcon, ViewIcon, ViewOffIcon } from '@chakra-ui/icons';
+import {
+    faCircleUser,
+    faCloudArrowUp,
+    faFont,
+    faT,
+    faTableColumns,
+} from '@fortawesome/free-solid-svg-icons';
+
+const IconMap = {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/icons/icons-map.ts:10]: this is not Map, this is Enum!

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/components/common/password-input/password-input.tsx [hunk 7]
```diff
@@ -37,7 +37,14 @@ const PasswordInput: React.FC<Properties> = ({ label, name, hasError }) => {
...
                     }
-                    icon={isPasswordVisible ? <ViewIcon /> : <ViewOffIcon />}
+                    as={Icon}
+                    icon={
+                        isPasswordVisible ? (
+                            <Icon as={IconMap.VIEW} />
+                        ) : (
+                            <Icon as={IconMap.VIEW_OFF} />
+                        )
+                    }
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/auth/components/common/password-input/password-input.tsx:47]:
```suggestion
                    icon={<Icon as={isPasswordVisible ? IconMap.VIEW : IconMap.VIEW_OFF}                    }
```

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/components/common/password-input/password-input.tsx (hunk 7 above)
Comment (lfelix3011) [reply frontend/src/bundles/auth/components/common/password-input/password-input.tsx:47]: @anton-otroshchenko Thanks

**Review by o-nedashkivska — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**